### PR TITLE
[SPARK-38574][DOCS] Enrich the documentation of option avroSchema

### PR DESCRIPTION
**Pull request by tianhanhu**
### What changes were proposed in this pull request?
Enrich Avro data source documentation to emphasize the difference between
`avroSchema` which is an option, and `jsonFormatSchema` which is a parameter to function `from_avro` .

When using `from_avro`, `avroSchema` option can be set to a compatible and evolved schema, while `jsonFormatSchema` has to be the actual schema. Elsewise, the behavior is undefined.

### Why are the changes needed?
Reduce confusion caused by option and parameter having similar namings.

### Does this PR introduce _any_ user-facing change?
Yes, Avro data source documentation is enriched a bit.

### How was this patch tested?
No testing required. Just a documentation change

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @gengliangwang FYI

**Review by xuanyuanking — APPROVED** [review]
LGTM
Will build the doc to double check the change.

File: docs/sql-data-sources-avro.md [hunk 1]
```diff
@@ -234,7 +234,9 @@ Data source options of Avro can be set via:
           When reading Avro, this option can be set to an evolved schema, which is compatible but different with
           the actual Avro schema. The deserialization schema will be consistent with the evolved schema.
           For example, if we set an evolved schema containing one additional column with a default value,
-          the reading result in Spark will contain the new column too.
+          the reading result in Spark will contain the new column too. Note that when using this option with 
+          <code>from_avro</code>, you still need to pass the actual schema as a parameter to the function. 
+          Otherwise, the behavior is undefined: it may fail or return arbitrary result.
```
Comment (gengliangwang) [inline docs/sql-data-sources-avro.md:239]: The actual schema is required by the from_avro API. Thus, the "otherwise" case won't happen.
I suggest we explain that the jsonSchema is for the actual input schema, while the "avroSchema" is for the output schema

**Review by weixiuli — APPROVED** [review]

File: docs/sql-data-sources-avro.md (hunk 1 above)
Comment (tianhanhu) [reply docs/sql-data-sources-avro.md]: Make sense. Removed the "otherwise" clause.
I think emphasizing on the difference between "actual" and "evolved" should be enough for this case? Further mentioning input vs output seems a bit wordy for me... @gengliangwang

File: docs/sql-data-sources-avro.md [hunk 2]
```diff
@@ -234,7 +234,8 @@ Data source options of Avro can be set via:
           When reading Avro, this option can be set to an evolved schema, which is compatible but different with
```
Comment (gengliangwang) [inline docs/sql-data-sources-avro.md:234]:
```suggestion
          When reading Avro files or calling function <code>from_avro</code>, this option can be set to an evolved schema, which is compatible but different with
```

Comment (gengliangwang) [reply docs/sql-data-sources-avro.md:234]: In line 247 we mention "read, write and function <code>from_avro</code>". Let's mention the function call here too.

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
verified on local setup:
image

**Comment by gengliangwang** [comment]
Merging to master/3.3
